modules/http4s/src/smithy4s/http4s/SimpleRestJsonBuilder.scala

```diff
@@ -19,11 +19,8 @@ package http4s
 
 import smithy4s.internals.InputOutput
 
-object SimpleRestJsonBuilder extends SimpleRestJsonBuilder(1024) {
-  def withMaxArity(maxArity: Int): SimpleRestJsonBuilder =
-    new SimpleRestJsonBuilder(maxArity) {}
-}
-abstract class SimpleRestJsonBuilder(maxArity: Int)
+object SimpleRestJsonBuilder extends SimpleRestJsonBuilder(1024)
+case class SimpleRestJsonBuilder(maxArity: Int)
```

Comment:
let's go back to a normal class + the `withMaxArity` method. Case classes are notoriously bad for binary compatibility (see [this](https://github.com/scala/improvement-proposals/pull/50) for instance) and this builder is something very public-facing.

Reply:
thanks for the link, I forgot about that SIP!

Reply:
Thanks, will take a look at that